kuma/javascript/src/section-link.jsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { render } from 'react-dom';
+
+import { gettext } from './l10n.js';
+
+import Anchor from './icons/anchor.svg';
+
+/**
+ * Returns a section anchor element with a nested SVG icon
+ * @param {Object} heading - The current heading DOM node
+ * @returns section anchor element as an HTML element
+ */
+export default function sectionAnchor(heading) {
+    const anchor = document.createElement('a');
+    anchor.href = `#${heading.id}`;
+    anchor.classList.add('section-link');
+    // heading.innerText is the current heading and will already be translated
+    anchor.setAttribute(
+        'aria-label',
+        `${gettext('Link to')} ${heading.innerText}`
+    );
+    // render the SVG into the anchor element
+    render(<Anchor />, anchor);
```

Comment:
This seems scary to me. Since this is being called from an effect function that is touching up pre-defined HTML, it would be cleaner if we could do it without react.  I don't know if this is a performance issue or not, but I think that the only other place we call render() is at the toplevel in index.jsx.

I think it would be cleaner to define the svg icon as a string constant in this file and then use innerHTML to inject it into the anchor.  Or maybe better use innerHTML to define it the first time in an off-screen element, and then call cloneNode(true) to make a copy each time you need another icon.

Reply:
Both Schalk and I tried to find a reason not to use `render()`. It worked but it was hard to figure out if it was dangerous or slow. It's quite possibly async and relatively fast so you can't just...

```javascript
console.time('render');
render(<Anchor />, anchor);
console.timeEnd('render');
```
since it'll always be a tiny number and there's a chance there's something async and deep. 

Makes me wonder, it is perhaps better to log how often the article re-renders after a hydration and see if that number changes depending on this line.